src/ir_Argo.cpp

```diff
@@ -505,4 +1748,98 @@ bool IRrecv::decodeArgo(decode_results *results, uint16_t offset,
   // is a union data type.
   return true;
 }
+
+/// Decode the supplied Argo message (WREM3).
+/// Status: Confirmed working w/ Argo 13 ECO (WREM-3)
+/// @param[in,out] results Ptr to the data to decode & where to store the decode
+///   result.
+/// @param[in] offset The starting index to use when attempting to decode the
+///   raw data. Typically/Defaults to kStartOffset.
+/// @param[in] nbits The number of data bits to expect.
+/// @param[in] strict Flag indicating if we should perform strict matching.
+/// @return A boolean. True if it can decode it, false if it can't.
+/// @note This decoder is separate from @c decodeArgo to maintain backwards
+///       compatibility. Contrary to WREM2, this expects a footer and gap!
+bool IRrecv::decodeArgoWREM3(decode_results *results, uint16_t offset,
+                             const uint16_t nbits,
+                             const bool strict) {
+  if (strict
+      && nbits != kArgo3AcControlStateLength * 8
+      && nbits != kArgo3ConfigStateLength * 8
+      && nbits != kArgo3iFeelReportStateLength * 8
+      && nbits != kArgo3TimerStateLength * 8) {
+    return false;
+  }
+
+  const uint16_t kArgoOverhead = 3;
+  uint16_t bytesRead = matchGeneric(results->rawbuf + offset, results->state,
+                  results->rawlen - offset, nbits,
+                  kArgoHdrMark, kArgoHdrSpace,
+                  kArgoBitMark, kArgoOneSpace,
+                  kArgoBitMark, kArgoZeroSpace,
+                  kArgoBitMark, kArgoGap,  // difference vs decodeArgo
+                  true, _tolerance, kArgoOverhead,
+                  false);
+  if (!bytesRead ||
+      !IRArgoAC_WREM3::isValidWrem3Message(results->state, nbits, strict)) {
+    return false;
+  }
+
+  // Success: Matched ARGO protocol and WREM3-model
+  // Note that unfortunately decode_type does not allow to persist model...
+  // so we will be re-detecting it later :)
+  results->decode_type = decode_type_t::ARGO;
+  results->bits = nbits;
+  // No need to record the state as we stored it as we decoded it.
+  // As we use result->state, we don't record value, address, or command as it
+  // is a union data type.
+  return true;
+}
+
+/// @brief Detects if an ARGO protocol message is a WREM-3 sub-type (model)
+/// @param state The raw IR decore state
+/// @param nbits The length of @c state **IN BITS**
+/// @param strict Whether to perform strict matching (incl. checksum)
+/// @return True if the message is a WREM-3 one
+bool IRArgoAC_WREM3::isValidWrem3Message(const uint8_t state[],
+                                         const uint16_t nbits, bool strict) {
+  auto stateLength = std::min(static_cast<uint16_t>(std::ceil(nbits / 8.0)),
+                              kStateSizeMax);
+
+  if (strict && !IRArgoAC_WREM3::hasValidPreamble(state, stateLength)) {
+    return false;
+  }
+
+  auto messageType = IRArgoACBase<ArgoProtocolWREM3>::getMessageType(state,
```

Comment:
Got it, will make explicit.
Not sure if I get the comment though - i.e. `messageType` is of an enum type, so it hardly violates `All internal library code must use [c99 exact-width type definitions](https://en.wikipedia.org/wiki/C_data_types#Fixed-width_integer_types). e.g. uint16_t instead of unsigned int etc.` 

----
Just in case, I'll remove all uses of `auto`, but let me add the following (opinioated!) thoughts :) :
- If you'd like to avoid `auto` entirely in the core library, I strongly suggest adding a new note to the code guidelines, banning its use **explicitly**.
- It's a heavily opinioated subject, but FYI, I'm in the "_use of auto is often **superior** to writing explicit types as it allows the compiler to do its magic_" camp. See also: https://herbsutter.com/2013/06/13/gotw-93-solution-auto-variables-part-2/. Aka. I am removing it, but with a heavy heart :)

Reply:
Re: Adding note. I will. Thanks for the suggestion.

Oh I get using `auto` is much better, but to better simulate what's happening on an ESP platform, I need ensure the bit-lengths of integers. We've had this bite more than once. i.e. `int` being 16 bits on the platform and 32 on x86 land etc.
`auto`s should be fine for a simple iterator that isn't used in any of the loop's body for bit arithmatic. It's just safest to not have it so the unit tests pick up problems vs. compiler optimisations. I wish I could have both worlds. :-(

Reply:
Just for the record - I fully get and embrace the need for having fixed-size integers on any ABI or HW interface. 
The comment you added though was related to a fully qualified fixed size enum (referred to via auto), which is what I reacted to :)
 In any case, it's gone now, so I think we can safely close this side-topic :)